test/test_grammars.rb

```diff
@@ -63,47 +80,35 @@ def test_local_scopes_are_in_sync
   end
 
   def test_submodules_have_licenses
-    categories = submodule_paths.group_by do |submodule|
-      files = Dir[File.join(ROOT, submodule, "*")]
-      license = files.find { |path| File.basename(path) =~ /\b(un)?licen[cs]e\b/i } || files.find { |path| File.basename(path) =~ /\bcopying\b/i }
-      if license.nil?
-        if readme = files.find { |path| File.basename(path) =~ /\Areadme\b/i }
-          license = readme if File.read(readme) =~ /\blicen[cs]e\b/i
-        end
-      end
-      if license.nil?
-        :unlicensed
-      elsif classify_license(license)
-        :licensed
+    submodule_licenses.each do |submodule, license|
+      next if PROJECT_WHITELIST.include?(submodule)
+
+      license_file = Licensee::Project.new(submodule).license_file
+      submodule_name = submodule.split("/").last
+
+      if license_file
+        assert license, "Submodule #{submodule_name} contains an unrecognized license. Please update #{__FILE__} to recognize the license"
```

Comment:
The existing test would print out all errors at once. The way the test is currently written it will print out the very first error and stop. It'd be nice to keep showing all errors like we did before.